repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -114,3 +145,37 @@ def check_file_modification(config):
     """
     output = _read_rpm_modifications(config)
     return _parse_config_modification(output, config)
+
+
+def get_leapp_packages(source_version=None, component=None, include_deps=False):
+    """
+    Get list of leapp packages.
+
+    :param source_version: a list or string specifying source_versions. If not defined then all packages
+        for current source_target_version will be returned.
+    :param component: a list or string specifying leapp components. If defined then only packages related
+        to the specific component will be returned.
+    :param include_deps: a flag to control whether deps meta packages are to be filtered out.
+    """
+    res = set()
+    source_versions = [source_version] if isinstance(source_version, str) else source_version
+    if not source_versions:
+        # No source_version of interest specified -> treat as if only current source system version
+        # requested
+        source_versions = [get_source_major_version()]
+    components = [component] if isinstance(component, str) else component
+    if not component:
+        # No component of interest specified -> treat as if packages for all components are requested
+        components = LEAPP_DEFAULT_COMPONENTS
```

Comment:
The comment and code are actually in conflict. Currently it does not return list of all packages or all components, but default components only. Cockpit-leapp is not included now. Hence the name of the constant is to me confusing.

Reply:
`LEAPP_DEFAULT_COMPONENTS` mean exactly what default components should - leapp components that are returned by default unless any specific ones have been requested.
I'll update the comment and rename the constant to `_LEAPP_DEFAULT_COMPONENTS`, no problem.

Reply:
I agree with the suggested solution (update comment + make it private entity).

I do not agree  that "default" is clear, because the current implementation does not corerspond to exactly what default means. Currently, the default contains also `snactor` (tools), which does not have to be installed in the system to do the upgrade. Default from my understanding would be set of packages that are installed when user run `dnf install leapp-upgrade` - iow mandatory packages. Snactor is not part of it. Having it defined publicly, it's not saying it's related to the function - for that, it would require docstring saying that - which in case of public/shared stuff, docstring should be always present. Having it private resolves this problem.

Reply:
When I read `DEFAULT` my understanding is that it is the default for the component argument to the `get_leapp_packages()` function.  But I agree that there's more than one meaning of default which the user could be conscious of here.  Maybe `LEAPP_*COMPONENTS`  fails to link `DEFAULT` to the function?  So something like `GET_LEAPP_PACKAGES_COMPONENT_DEFAULT` is better?  (But really long).

Note: I think the discussion about what to name the parameter also leads to this confusion: we chose to name it `component` because there is a pre-existing LEAPP concept of `component`.  But now, when we name this variable, it is only about the parameter name, not about the larger concept that the parameter name refers to.